core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.ts

```diff
@@ -0,0 +1,156 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Backend api service for fetching and resolving suggestions.
+ */
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { FetchSuggestionsResponse } from './contribution-and-review.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+
+interface ResolveToExplorationData {
+  action: string;
+  'review_message': string;
+  'commit_message': string;
+}
+
+interface ResolveToSkillData {
+  action: string;
+  'review_message': string;
+  'skill_difficulty': string;
+}
+
+interface UpdateTranslationData {
+  'translation_html': string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewBackendApiService {
+  private _SUBMITTED_SUGGESTION_LIST_HANDLER_URL = (
+    '/getsubmittedsuggestions/<target_type>/<suggestion_type>');
+  private _REVIEWABLE_SUGGESTIONS_HANDLER_URL = (
+    '/getreviewablesuggestions/<target_type>/<suggestion_type>');
+  private _SUGGESTION_TO_EXPLORATION_ACTION_HANDLER_URL = (
+    '/suggestionactionhandler/exploration/<exp_id>/<suggestion_id>');
+  private _SUGGESTION_TO_SKILL_ACTION_HANDLER_URL = (
+    '/suggestionactionhandler/skill/<skill_id>/<suggestion_id>');
+  private _UPDATE_TRANSLATION_HANDLER_URL = (
+    '/updatetranslationsuggestionhandler/<suggestion_id>');
+  private _UPDATE_QUESTION_HANDLER_URL = (
+    '/updatequestionsuggestionhandler/<suggestion_id>');
+
+  private SUBMITTED_QUESTION_SUGGESTIONS = (
+    'SUBMITTED_QUESTION_SUGGESTIONS');
+  private REVIEWABLE_QUESTION_SUGGESTIONS = (
+    'REVIEWABLE_QUESTION_SUGGESTIONS');
+  private SUBMITTED_TRANSLATION_SUGGESTIONS = (
+    'SUBMITTED_TRANSLATION_SUGGESTIONS');
+  private REVIEWABLE_TRANSLATION_SUGGESTIONS = (
+    'REVIEWABLE_TRANSLATION_SUGGESTIONS');
+
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  async fetchSuggestionsAsync(
+      fetchType: string,
+  ): Promise<FetchSuggestionsResponse> {
+    if (fetchType === this.SUBMITTED_QUESTION_SUGGESTIONS) {
+      return this.fetchSubmittedSuggestionsAsync('skill', 'add_question');
+    } else if (fetchType === this.SUBMITTED_TRANSLATION_SUGGESTIONS) {
+      return this.fetchSubmittedSuggestionsAsync(
+        'exploration', 'translate_content');
+    } else if (fetchType === this.REVIEWABLE_QUESTION_SUGGESTIONS) {
+      return this.fetchReviewableSuggestionsAsync('skill', 'add_question');
+    } else if (fetchType === this.REVIEWABLE_TRANSLATION_SUGGESTIONS) {
+      return this.fetchReviewableSuggestionsAsync(
+        'exploration', 'translate_content');
+    }
```

Comment:
```suggestion
    if (fetchType === this.SUBMITTED_QUESTION_SUGGESTIONS) {
      return this.fetchSubmittedSuggestionsAsync('skill', 'add_question');
    }
    if (fetchType === this.SUBMITTED_TRANSLATION_SUGGESTIONS) {
      return this.fetchSubmittedSuggestionsAsync(
        'exploration', 'translate_content');
    }
    if (fetchType === this.REVIEWABLE_QUESTION_SUGGESTIONS) {
      return this.fetchReviewableSuggestionsAsync('skill', 'add_question');
    }
    if (fetchType === this.REVIEWABLE_TRANSLATION_SUGGESTIONS) {
      return this.fetchReviewableSuggestionsAsync(
        'exploration', 'translate_content');
    }
```

You don't need else here because of the return statement

Reply:
done